### PR TITLE
Add a basic schema and tests

### PR DESCRIPTION
**Pull request by anurag90x**
This covers -

1. Paragraphs
2. Headlines
3. Lists
4. Images and links
5. Text formatting

There are tests to check successful schema validation as well as xss
checks.

### CODE REVIEW

File: schema.xsd [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      <xs:length value="0"/>
+    </xs:restriction>
+  </xs:simpleType>
+
+  <xs:complexType name="imageTag">
+    <xs:attribute name="width" use="optional" type="xs:string"/>
+    <xs:attribute name="height" use="optional" type="xs:string"/>
+    <xs:attribute name="alt" use="optional" type="xs:string"/>
+    <xs:attribute name="class" use="optional" type="xs:string"/>
+    <xs:attribute ref="src"/>
```
Comment (LeenaMansour) [inline schema.xsd]: Alignment?

File: schema.xsd [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  </xs:complexType>
+
+  <xs:complexType name="linkTag" mixed="true">
+    <xs:choice minOccurs="0" maxOccurs="1">
+      <xs:group ref="formattedTextGroup"/>
+    </xs:choice>
+    <xs:attribute ref="href"/>
+    <xs:attribute name="title" use="optional" type="xs:string"/>
```
Comment (LeenaMansour) [inline schema.xsd]: Does that mean the name will always be title?

File: markup_language/schema_validator.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+def load_schema():
+    dir = os.path.dirname(__file__)
+    file_name = os.path.join(dir, '../schema.xsd')
+    with open(file_name, 'r') as f:
+        schema_root = etree.XML(f.read())
+        return etree.XMLSchema(schema_root)
+
+
+def add_closing_tags(match):
+    return match.group(0).replace(">", "/>")
```
Comment (markstory) [inline markup_language/schema_validator.py]: Clever.

File: schema.xsd [hunk 4]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.1"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+
+  <xs:attribute name="src">
+    <xs:simpleType>
+      <xs:restriction base="xs:string">
+        <xs:pattern value="https:\S+"/>
```
Comment (markstory) [inline schema.xsd]: Don't we also want to lock this to our domains?

File: schema.xsd [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      </xs:restriction>
+    </xs:simpleType>
+  </xs:attribute>
+
+  <xs:attribute name="href">
+    <xs:simpleType>
+      <xs:restriction base="xs:string">
+        <xs:pattern value="https:\S+"/>
```
Comment (markstory) [inline schema.xsd]: Won't people want to link to non secure domains?

File: schema.xsd [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      </xs:restriction>
+    </xs:simpleType>
+  </xs:attribute>
+
+  <xs:group name="headlineGroup">
+    <xs:choice>
+      <xs:element name="h1" type="textContainer"/>
+      <xs:element name="h2" type="textContainer"/>
+      <xs:element name="h3" type="textContainer"/>
+      <xs:element name="h4" type="textContainer"/>
```
Comment (markstory) [inline schema.xsd]: This precludes the use of inline formatting in headings but I see that as a good thing.

File: schema.xsd [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  <xs:complexType name="imageTag">
+    <xs:attribute name="width" use="optional" type="xs:string"/>
+    <xs:attribute name="height" use="optional" type="xs:string"/>
+    <xs:attribute name="alt" use="optional" type="xs:string"/>
+    <xs:attribute name="class" use="optional" type="xs:string"/>
+    <xs:attribute ref="src"/>
+  </xs:complexType>
+
+  <xs:complexType name="linkTag" mixed="true">
+    <xs:choice minOccurs="0" maxOccurs="1">
```
Comment (markstory) [inline schema.xsd]: You might want more than 1. That would allow bold & italic segments inside a link.

File: schema.xsd [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:element name="div">
+    <xs:complexType>
+      <xs:choice minOccurs="0" maxOccurs="unbounded">
+        <xs:group ref="headlineGroup"/>
+        <xs:group ref="listGroup"/>
+        <xs:element name="p" type="textContainer"/>
+        <xs:element name="a" type="linkTag"/>
```
Comment (markstory) [inline schema.xsd]: Do we want to allow `<br>` in the document root?

File: tests/documents/paragraphs_with_images_xss.html [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<div>
+  <p>
+    <img class="fr-fir fr-dii fr-draggable" width="200" src="https://cdn0.froala.com/assets/editor/pages/B/editor-photo-9fb6e5149212f06a0a54df52ccc30934.jpg" alt="Editor photo">
+    <img class="fr-fir fr-dii fr-draggable" width="400" src="https://cdn0.froala.com/assets/editor/pages/B/editor-photo-9fb6e5149212f06a0a54df52ccc30934.jpg" alt="New image yay">
```
Comment (markstory) [inline tests/documents/paragraphs_with_images_xss.html]: Is it worth having a test for non http URLs in images?

**Comment by AntonNguyen** [comment]
Does the anchor tag accept the `title` attribute?

File: tests/documents/paragraphs_with_images_xss.html (hunk 9 above)
Comment (anurag90x) [inline tests/documents/paragraphs_with_images_xss.html]: There's one below with a javascript in the src.

Comment (markstory) [inline tests/documents/paragraphs_with_images_xss.html]: Cool.

File: schema.xsd (hunk 6 above)
Comment (anurag90x) [inline schema.xsd]: Yeah none of the elements handle any inline styles.

File: schema.xsd (hunk 7 above)
Comment (anurag90x) [inline schema.xsd]: You right.

File: schema.xsd (hunk 8 above)
Comment (anurag90x) [inline schema.xsd]: We should so I made that change.

**Pull request merged**